libraries/standard/utilities/include/iot_logging_setup.h

```diff
@@ -0,0 +1,289 @@
+ * @endcode
+ */
+
+/**
+ * @def IotLogInfo( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_INFO.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_INFO, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogInfoWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_INFO.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_INFO, ... )
+ * @endcode
+ */
+
+/**
+ * @def IotLogDebug( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_DEBUG.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_DEBUG, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogDebugWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_DEBUG.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_DEBUG, ... )
+ * @endcode
+ */
+
+#if !defined( IotLog )
+    #error "Please define IotLog logging interface."
+#endif
+
+/* Check that LIBRARY_LOG_LEVEL is defined and has a valid value. */
+#if !defined( LIBRARY_LOG_LEVEL ) ||           \
+    ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) &&  \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) )
+    #error "Please define LIBRARY_LOG_LEVEL as either IOT_LOG_NONE, IOT_LOG_ERROR, IOT_LOG_WARN, IOT_LOG_INFO, or IOT_LOG_DEBUG."
+/* Check that LIBRARY_LOG_NAME is defined and has a valid value. */
+#elif !defined( LIBRARY_LOG_NAME )
+    #error "Please define LIBRARY_LOG_NAME."
+#else /* if !defined( LIBRARY_LOG_LEVEL ) || ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) ) */
+    #if LIBRARY_LOG_LEVEL >= IOT_LOG_DEBUG
+        /* All log levels will logged. */
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )                  IotLog( IOT_LOG_WARN, "%s", pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )     IotLog( IOT_LOG_WARN, pFormat, __VA_ARGS__ )
+        #define IotLogInfo( pFormat )                  IotLog( IOT_LOG_INFO, "%s", pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )     IotLog( IOT_LOG_INFO, pFormat, __VA_ARGS__ )
+        #define IotLogDebug( pFormat )                 IotLog( IOT_LOG_DEBUG, "%s", pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )    IotLog( IOT_LOG_DEBUG, pFormat, __VA_ARGS__ )
+
+/* If log level is DEBUG, enable the function to print buffers. */
+        #define IotLog_PrintBuffer( pHeader, pBuffer, bufferSize ) \
+    IotLog_GenericPrintBuffer( LIBRARY_LOG_NAME,                   \
```

Comment:
Alignment again - Is it better to run uncrusitify?

Reply:
Don't have uncrustify.cfg file on this branch. So my format on save IDE feature wasn't working 😛 . 
Will format them (and add the uncrustify file to this branch as well)